### PR TITLE
[JS] Topic: Closures in JavaScript

### LINKED ISSUE #1265: [JS]: Closures in Javascript
<hr>

## Description 📜

- What are closures?
- How closures work?
- What are the advantages of closures?
- What are the Disadvantages of closures?
 
<hr>

## Domain of Contribution 📊

- [x] JavaScript
<hr>

### PR DESCRIPTION
**Pull request by Ananyagupta43**
<hr>

## Description 📜
Created a documentation on Closures in JavaScript .The Documentation explains the concepts of execution context,lexical environment and working of closures.

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #1265
<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] JavaScript

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review.

File: JavaScript/Topics/12.Closures_In_JavaScript/Readme.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+- Function like once(it is a function which can run only once) uses closures.
+- Concept of closures in used in module design pattern.
+  
+
+# Disadvantages Of Closures
+- Over consumption of memory as the variables are not garbage collected properly.
+- Closures can cause memory leaks if they are handled properly .
+
+
+### Contributor : [Ananya Gupta](https://github.com/Ananyagupta43)
```
Comment (dakshsethi) [inline JavaScript/Topics/12.Closures_In_JavaScript/Readme.md:124]: Doesn't need to be a heading
Just write this 
`Contributor : [Ananya Gupta](https://github.com/Ananyagupta43)`

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/12.Closures_In_JavaScript/Readme.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ >If we go by definition, then closure is the combination of a function bundled together with its lexical environment.
+<p>&nbsp;</p>
+
+ ## **Understanding Lexical Environment And Execution Context**
+ 
+Lexical Environment is basically a data structure that holds identifier-variable mapping. In simple words, lexical environment is the local memory along with the lexical environment of its parent. For understanding Lexical Environment we need to first understand ```Execution Context```.
+<p>&nbsp;</p>
+
+### **Execution Context**
+Execution context is an abstract space where the JavaScript code is  evaluated and executed.
```
Comment (Creatoon) [inline JavaScript/Topics/12.Closures_In_JavaScript/Readme.md:17]:
```suggestion
Execution context is an abstract space where the JavaScript code is evaluated and executed.
```

File: JavaScript/Topics/12.Closures_In_JavaScript/Readme.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+Here the scope of variable q is inside the function p, so q can be accessed by any function inside function p, function r is lexically inside function p and p is the parent of r.
+
+
+So for the console statement the JS Engine will search the value of q inside the local memory of r but if q is not present inside r then the JS Engine will search the value in the lexical environment of its parent (here the parent is p). This chain of searching variables is called `Scope Chain`.
+
+<p>&nbsp;</p>
+
+## Working of Closures
+
+We have two functions c and a:=
```
Comment (Creatoon) [inline JavaScript/Topics/12.Closures_In_JavaScript/Readme.md:73]:
```suggestion
We have two functions c and a :=
```

**Review by Creatoon — APPROVED** [review]

**Comment by Creatoon** [comment]
@Ananyagupta43 Just make above few changes , rest i approved the pr as other things looks good to me and these are just minor mistakes

**Comment by Ananyagupta43** [comment]
@Creatoon Ok sir I'll make these changes.

**Comment by Ananyagupta43** [comment]
@Creatoon Sir I made the changes .Please review it.

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

**Comment by Ananyagupta43** [comment]
@dakshsethi Thanks Sir.

**Comment by Ananyagupta43** [comment]
@paramsiddharth Sir I have made all the necessary changes please review it.

**Pull request merged**